sql/core/src/main/scala/org/apache/spark/sql/catalog/Catalog.scala

```diff
@@ -342,6 +342,44 @@ abstract class Catalog {
     createTable(tableName, source, schema, options)
   }
 
+  /**
+   * Creates a table based on the dataset in a data source and a set of options.
+   * Then, returns the corresponding DataFrame.
+   *
+   * @param tableName is either a qualified or unqualified name that designates a table.
+   *                  If no database identifier is provided, it refers to a table in
+   *                  the current database.
+   * @since 3.1.0
+   */
+  def createTable(
+      tableName: String,
+      source: String,
+      options: java.util.Map[String, String],
+      description: String): DataFrame = {
```

Comment:
@nchammas, could we place `option` at the last?

Reply:
Can do, but just so I understand: Consistent with what?

Reply:
I mean consistency with other APIs above which places `options` at the last.